src/panels/lovelace/cards/hui-statistic-card.ts

```diff
@@ -0,0 +1,299 @@
+    this._fetchMetadata();
+
+    if (this._config.footer) {
+      this._footerElement = createHeaderFooterElement(this._config.footer);
+    } else if (this._footerElement) {
+      this._footerElement = undefined;
+    }
+  }
+
+  public async getCardSize(): Promise<number> {
+    let size = 2;
+    if (this._footerElement) {
+      const footerSize = computeCardSize(this._footerElement);
+      size += footerSize instanceof Promise ? await footerSize : footerSize;
+    }
+    return size;
+  }
+
+  protected render(): TemplateResult {
+    if (!this._config || !this.hass) {
+      return html``;
+    }
+
+    if (this._error) {
+      return html` <ha-alert alert-type="error">${this._error}</ha-alert> `;
+    }
+
+    const stateObj = this.hass.states[this._config.entity];
+    const name =
+      this._config.name ||
+      getStatisticLabel(this.hass, this._config.entity, this._metadata);
+
+    return html`
+      <ha-card @click=${this._handleClick} tabindex="0">
+        <div class="header">
+          <div class="name" .title=${name}>${name}</div>
+          <div class="icon">
+            <ha-state-icon
+              .icon=${this._config.icon}
+              .state=${stateObj}
+            ></ha-state-icon>
+          </div>
+        </div>
+        <div class="info">
+          <span class="value"
+            >${this._value === undefined
+              ? ""
+              : this._value === null
+              ? "?"
+              : formatNumber(this._value, this.hass.locale)}</span
+          >
+          <span class="measurement"
+            >${this._config.unit ||
+            getDisplayUnit(
+              this.hass,
+              this._config.entity,
+              this._metadata
+            )}</span
+          >
+        </div>
+        ${this._footerElement}
+      </ha-card>
+    `;
+  }
+
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    // Side Effect used to update footer hass while keeping optimizations
+    if (this._footerElement) {
+      this._footerElement.hass = this.hass;
+    }
+    if (
+      changedProps.has("_value") ||
+      changedProps.has("_metadata") ||
+      changedProps.has("_error")
+    ) {
+      return true;
+    }
+    if (this._config) {
+      return hasConfigOrEntityChanged(this, changedProps);
+    }
+    return true;
+  }
+
+  protected firstUpdated() {
+    this._fetchStatistic();
+    this._fetchMetadata();
+  }
+
+  protected updated(changedProps: PropertyValues) {
+    super.updated(changedProps);
+    if (!this._config || !this.hass) {
+      return;
+    }
+
+    const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
+    const oldConfig = changedProps.get("_config") as
+      | EntityCardConfig
+      | undefined;
+
+    if (
+      !oldHass ||
+      !oldConfig ||
+      oldHass.themes !== this.hass.themes ||
+      oldConfig.theme !== this._config.theme
+    ) {
+      applyThemesOnElement(this, this.hass.themes, this._config!.theme);
+    }
```

Comment:
Just a note for future. Do we want to keep this theme feature at the card level ?

Reply:
I think so?